lib/schema.js

```diff
@@ -213,10 +213,24 @@ Schema.prototype.refKeys = function(mdl) {
   for (var i = 0; i < refIndices.length; ++i) {
     var refIndex = refIndices[i];
     var refValues = [];
+    // This value will remain false unless a relevant value is found
+    var defined = false;
     for (var j = 0; j < refIndex.fields.length; ++j) {
-      refValues.push(this.fieldVal(mdl, refIndex.fields[j]));
+      var value = this.fieldVal(mdl, refIndex.fields[j]);
+      refValues.push(value);
+      //Prevents refdoc keys from having "undefined"
+      if(value){
```

Comment:
In fact, this is likely 'more correct', since in your case, you could potentially have an index of 2 fields, where one of them is undefined and one is defined.  In your case this would create a refdoc of `undefined-XXX`, whereras mine would do whats probably more expected, which is to skip the index entirely.